decidim/decidim
# Add Attachments with a Link

#### :tophat: What? Why?
Allows creating attachments with a link. 

- Link attachments won't have an upload (file) but a URL that will be stored in a new "link" column in the attachments table. 
- Link attachments will have a special `content_type` "decidim/link".
- Link attachments will have 0 `file_size`
- Link attachments can only be managed from the admin panel. 

#### :pushpin: Related Issues
- Fixes #12561

#### Testing
- Create link attachments from the admin panel.
- Test the different places where we display attachments and make sure these link attachments are displayed correctly. 
- Make sure all the rest of the places where you can create attachments work correctly. 

Staging links for spaces:
- [Admin > Process > Attachments](http://localhost:3000/admin/participatory_processes/authority-publish/attachments)
- [Process > Related documents](http://localhost:3000/processes/authority-publish)
- [Admin > Assemblies > Attachments](https://decidim-lot2.populate.tools/admin/assemblies/test/attachments)
- [Assembly > Related documents](https://decidim-lot2.populate.tools/assemblies/test)
- Initiatives
- Conferences

Staging links for components:
- [Admin > Meeting > Attachments](https://decidim-lot2.populate.tools/admin/participatory_processes/branch-highway/components/5/manage/meetings/5/attachments)
- [Meeting > Related documents](https://decidim-lot2.populate.tools/processes/branch-highway/f/5/meetings/5)
- [Admin > Accountability Results > Attachments](https://decidim-lot2.populate.tools/admin/participatory_processes/branch-highway/components/1/manage/results/1/attachments)
- [Accountability Results > Documents](https://decidim-lot2.populate.tools/processes/branch-highway/f/1/results/1)
- Budgets
- Proposals

### :camera: Screenshots

<img width="837" alt="Screenshot 2024-05-31 at 13 16 31" src="https://github.com/decidim/decidim/assets/935744/793386a5-a65c-4885-b0c0-956758679cfc">

<img width="951" alt="Screenshot 2024-05-31 at 13 17 16" src="https://github.com/decidim/decidim/assets/935744/89244369-2fc2-4936-a356-307a3aecd28e">
